### DISCUSSION TITLE
Main branch vs crates.io version

### DISCUSSION THREAD

**Discussion by afonsolage** [bevyengine/bevy#2574] [General]
So, I've been following Bevy since 0.1 and made some pongs in the past. Now I'm playing again, just for fun, I'm not intended to do serious business. So after reading recent commits and some discussions on GitHub Discussions I'm wondering if I should got for main branch, which has some considerable changes (and probably even more) or stick with 0.5 on crates.io and wait for 0.6 (which should take a while).

Most of the time I read the examples on repo, and since it is been kept update, this shouldn't be a problem. I wonder what you guys are doing, 0.5 on crates.io or main and if there is some serious problem by using main that I can see.

**Comment by Nilirad** [comment]
Most of the discussions here and on Discord are about the main branch. That's where the majority of the community focuses on. Even most third-party plugins depend on main branch.

It is quite safe to assume that every 0.x release until now feels outdated to the majority a couple of weeks after being published.

**Reply by dman-os** [reply]
> Even most third-party plugins depend on main branch.

Note that this isn't true for [bevy_rapier](https://github.com/dimforge/bevy_rapier) and [bevy_egui](https://github.com/mvlabat/bevy_egui/) if you were planning on depending on those.

**Reply by Nilirad** [reply]
There is a fork of [`bevy_rapier`](https://github.com/alec-deason/bevy_rapier/tree/tracking_changes_for_bevy_0.5) that target the main branch but, at least for the moment, the author seems focused on other tasks.

**Comment by marko-lazic** [comment]
I run from crate.io because I am concerned that my project code could potentially break when I actually want to do something productive. Then instead of working on what I wanted, I would have to spend hours, checking merged pull requests, figuring out how to fix my code. Then at the time I fixed the break I would already get tired from coding :).  When next version gets released the migration guide is already there so I have to migrate my code once. Having that said I always have another IDE opened with latest main branch, checking out examples reading new commits. If breaking changes do not concern you then I guess working from main would be fine.

**Reply by ypoonse** [reply]
I also think this is the best way of working, I'm doing the same. Unless if you depend on some new feature being developed on main I don't see a reason to chase the main branch, and prefer to migrate on controlled intervals and with a clear migration guide.
